### PR TITLE
extract without rimraf'ing the target folder

### LINKED ISSUE #24: Extract: rimraf dir contents, not dir itself
Right now, `Fetcher.extract` calls `Fetcher[_mkdir]` which [removes the dir before creating it](https://github.com/npm/pacote/blob/latest/lib/fetcher.js#L300-L320).

This code was copied [from pacote 9](https://github.com/npm/pacote/blob/v9/extract.js#L73-L74), and ostensibly ensures that the directory is empty before dumping package contents into it.

However, this is a problem if a consumer wants to extract a new version of a package over an existing node in the tree.

Here's a little script that'll tell pacote what to rimraf, if run with `depth: 1`.  Roll this into a new module and use it.  (`@npmcli/arborist` may need this for package reification as well.)  https://gist.github.com/isaacs/262f31fd9f37b25f27e4f74e20e2c491

### PR DESCRIPTION
**Pull request by isaacs**
Extract: rimraf dir contents, not dir itself

That is, without rm -rf'ing all the child nodes underneath it in the
tree.  This supports overwriting a package in the tree with a new
extraction, _without_ deleting all of the package's child nodes in the
node_module heirarchy.

This allows a better tree reification mechanism that does not trigger
EPERM and other folder removal/modification errors, especially on
Windows.

Fix #24

### CODE REVIEW

File: test/remote.js [hunk 1]
```diff
@@ -36,8 +36,7 @@ t.test('start server', t => {
       res.end(Buffer.alloc(2048))
     } else if (req.url === '/timeout') {
       res.statusCode = 200
-      res.write('just a second... ')
-      setTimeout(() => res.end('still there?'), 1000).unref()
+      // your call is important to us
```
Comment (mikemimik) [inline test/remote.js]: nit: not sure this is important?

File: lib/fetcher.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ const fsm = require('fs-minipass')
 const cacache = require('cacache')
 const osenv = require('osenv')
 const isPackageBin = require('./util/is-package-bin.js')
+const getContents = require('@npmcli/installed-package-contents')
```
Comment (mikemimik) [inline lib/fetcher.js]: nit: just making sure that this package has passing tests

File: test/remote.js (hunk 1 above)
Comment (isaacs) [reply test/remote.js]: It is, on travis 1/6th of the time, apparently :)

File: lib/fetcher.js (hunk 2 above)
Comment (isaacs) [reply lib/fetcher.js]: It does. https://github.com/npm/installed-package-contents/runs/351565160

File: test/remote.js (hunk 1 above)
Comment (mikemimik) [reply test/remote.js]: Perhaps a more verbose not for future folks reading through this because they got a failure?

**Pull request merged**